core/src/server/rpc_module.rs

```diff
@@ -750,7 +754,7 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 						claimed,
 					}));
 
-					callback(params, sink, ctx.clone());
+					let _ = callback(params, sink, ctx.clone());
```

Comment:
Ah I see; the result really isn't used at all!

I feel a bit uneasy about using a result and then throwing away the error; returning an error makes it feel like it'll be made use of in some way. We are just doing it because it plays nicely with the `?` type.

Perhaps we _should_ use a different error type in the response; something like `struct SubscriptionError` which hodls no data and impls From so that `?` can be used transparently on other jsonrpsee errors to convert them into it. This at least signals in the API typings that error information is being discarded when returned?

Other question; don't we want to use an error case here to make sure the connection is closed or something? For example if a subscription does something like:

```
        tokio::spawn(async move {
            pending.pipe_from_stream(params).await;
        });
        Err(SubscriptionError)
```

(or otherwise spawns a thread but still then decides to return an error), I'd expect the error response to lead to the connection being closed?

Reply:
would be nice with a comment here, I will likely forget about this when looking at the code in a few days :)

Reply:
Just want to confirm; do we not care about any explicit closing of the subscription here if an error is returned?

Reply:
I guess we could log it as `warn`.

Should only possible to fail if `params.parse()` or `SubscriptionEmptyError (connection was closed)`

Reply:
Have added a warning here with the `error` as well, just in case we decide to change it later to something else